adaptive_n_back/experiment.js

```diff
@@ -200,6 +234,9 @@ for (var b = 0; b < num_blocks; b++) {
 		};
 		adaptive_n_back_experiment.push(test_block)
 	}
+	if ($.inArray(b,[4,7,15]) != -1) {
```

Comment:
it's not i?

Looking at this, it seems that there are two things here - the choice to add attention checks, period, and the choice of what the "threshold" is for granting credit. But unlike a "reaction time" variable that is specific to an experiment, this seems like a general thing that could be added in any experiment, inserted every random N blocks (based on some criteria of the experiment). The same would be true of any kind of check,

With this current implementation idea, this attention check must be hardcoded into every experiment to be possible, and then things like the blocks where its added, or the details of the check, must be changed on the level of the experiment, That would be akin to hard coding css into 10 html files instead of using one common css file that each link to.

I think what we want instead is some kind of different thing, independent of the experiments, that can be dynamically added. It would be a choice the user makes when installing the experiment, and the checks would always be randomly inserted, the number of times based on some function of the experiment length/time. This means that the plugin file would be added to the page dynamically, and the blocks dynamically inserted either before the init is called or right at the beginning of the experiment, and we would need to test if we can do this (if the JsPsych variable is just a series of blocks, I don't see why we couldn't change them around right at the beginning, or add a snippet to the experiment.js before deployment.

Finally, it is occurring to me (as I think you mentioned) that basic stuffs like the number of experiment blocks should be available to the user as a variable to edit.  This would be very simple to do- we add them as experiment_variables (the "type" might be "experiment" or "run" instead of "bonus" or "credit" along with a "default" value, and given that the experiment is installed and run, the user given choice to set these variables.